WcaOnRails/spec/lib/auxiliary_data_computation_spec.rb

```diff
@@ -0,0 +1,143 @@
+      AuxiliaryDataComputation.compute_concise_results
+      # Concise single results
+      concise_single_results = ActiveRecord::Base.connection.execute "SELECT eventId, personId, year, best FROM ConciseSingleResults"
+      expect(concise_single_results).to match_array [
+        ["333", person.wca_id, 2016, 700],
+        ["333", person.wca_id, 2017, 800],
+        ["222", person.wca_id, 2017, 100],
+      ]
+      # Concise average results
+      concise_average_results = ActiveRecord::Base.connection.execute "SELECT eventId, personId, year, average FROM ConciseAverageResults"
+      expect(concise_average_results).to match_array [
+        ["333", person.wca_id, 2016, 800],
+        ["333", person.wca_id, 2017, 900],
+        ["222", person.wca_id, 2017, 150],
+      ]
+    end
+
+    it "creates multiple entries for people that have switched country in the middle of a year" do
+      FactoryGirl.create :result, eventId: "333", best: 700, average: 800, competition: competition_2016, person: person
+      person.update_using_sub_id! countryId: "Chile"
+      FactoryGirl.create :result, eventId: "333", best: 750, average: 850, competition: next_competition_2016, person: person
+      AuxiliaryDataComputation.compute_concise_results
+      # Concise single results
+      concise_single_results = ActiveRecord::Base.connection.execute "SELECT eventId, personId, countryId, year, best FROM ConciseSingleResults"
+      expect(concise_single_results).to match_array [
+        ["333", person.wca_id, "China", 2016, 700],
+        ["333", person.wca_id, "Chile", 2016, 750],
+      ]
+      # Concise average results
+      concise_average_results = ActiveRecord::Base.connection.execute "SELECT eventId, personId, countryId, year, average FROM ConciseAverageResults"
+      expect(concise_average_results).to match_array [
+        ["333", person.wca_id, "China", 2016, 800],
+        ["333", person.wca_id, "Chile", 2016, 850],
+      ]
+    end
+  end
+
+  describe ".compute_rank_tables", clean_db_with_truncation: true do
+    let(:australian) { FactoryGirl.create :person, countryId: "Australia" }
+    let(:canadian) { FactoryGirl.create :person, countryId: "Canada" }
+    let(:american_1) { FactoryGirl.create :person, countryId: "USA" }
+    let(:american_2) { FactoryGirl.create :person, countryId: "USA" }
+
+    def rank_333(person, ranks_type)
+      person.public_send(ranks_type).find_by(eventId: "333").attributes.symbolize_keys
+    end
+
+    before do
+      FactoryGirl.create :result, eventId: "333", best: 600, average: 700, person: australian
+      FactoryGirl.create :result, eventId: "333", best: 700, average: 800, person: american_1
+      FactoryGirl.create :result, eventId: "333", best: 800, average: 900, person: canadian
+      FactoryGirl.create :result, eventId: "333", best: 900, average: 1000, person: american_2
+    end
+
+    it "computes world, continental and national ranking position" do
```

Comment:
uber nit: I've been using the [serial comma](https://en.wikipedia.org/wiki/Serial_comma) in other places. It would be nice to be consistent about this.